internal/pkg/cli/local_run.go

```diff
@@ -292,23 +297,51 @@ func (o *localRunOpts) Execute() error {
 	}
 	o.imageInfoList = append(o.imageInfoList, sidecarImageLocations...)
 
-	err = o.runPauseContainer(context.Background(), containerPorts)
-	if err != nil {
-		return err
-	}
+	errCh := make(chan error, 1)
+	// Handle containers when a termination signal is received.
+	go func() {
+		sig := o.NotifyInterrupt()
+		fmt.Println("Received signal:", sig)
+		o.mutex.Lock()
+		o.isContainerTermination = true
+		o.mutex.Unlock()
+		err := o.handleContainers()
+
+		// Wait for a short duration to allow containers to stop or get killed.
+		time.Sleep(1 * time.Second)
+		errCh <- err
+	}()
+
+	go func() {
+		err = o.runPauseContainer(context.Background(), containerPorts)
+		if err != nil {
+			errCh <- err
+			return
+		}
+		err = o.runContainers(context.Background(), o.imageInfoList, secretsList, envVars)
+		if err != nil {
+			errCh <- err
+			return
+		}
+	}()
 
-	err = o.runContainers(context.Background(), o.imageInfoList, secretsList, envVars)
+	err = <-errCh
 	if err != nil {
 		return err
 	}
-
 	return nil
 }
 
 func (o *localRunOpts) getContainerSuffix() string {
 	return fmt.Sprintf("%s-%s-%s", o.appName, o.envName, o.wkldName)
 }
 
+func (o *localRunOpts) NotifyInterrupt() os.Signal {
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
why do we need a `SIGTERM` if you only want to capture `ctrl-c`?

Reply:
You are correct, if its only control c we don't require `SIGTERM`